behaviour/graql/language/undefine.feature

```diff
@@ -260,45 +1218,286 @@ Feature: Graql Undefine Query
+    Given graql insert
       """
-      undefine person sub entity;
+      insert $x isa person, has email "samuel@grakn.ai";
+      """
+    Given the integrity is validated
+    Given get answers of graql query
+      """
+      match
+        $x has name $n;
+      get $n;
+      """
+    Given concept identifiers are
+      |     | check | value       |
+      | SAM | value | name:Samuel |
+    Given uniquely identify answer concepts
+      | n   |
+      | SAM |
+    When graql undefine
+      """
+      undefine samuel-email-rule sub rule;
       """
     Then the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x has name $n;
+      get $n;
+      """
+    Then answer size is: 0
+
 
+  ############
+  # ABSTRACT #
+  ############
 
-  Scenario: undefine an entity type throws if instances exist
+  Scenario: undefining a type as abstract converts an abstract to a concrete type, allowing creation of instances
+    Given get answers of graql query
+      """
+      match
+        $x type abstract-type;
+        not { $x abstract; };
+      get;
+      """
+    Given answer size is: 0
+    Given graql insert throws
+      """
+      insert $x isa abstract-type;
+      """
+    Given the integrity is validated
+    Given graql undefine
+      """
+      undefine abstract-type abstract;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x type abstract-type;
+        not { $x abstract; };
+      get;
+      """
+    Then concept identifiers are
+      |     | check | value         |
+      | ABS | label | abstract-type |
+    Then uniquely identify answer concepts
+      | x   |
+      | ABS |
+    When graql insert
+      """
+      insert $x isa abstract-type;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match $x isa abstract-type; get;
+      """
+    Then answer size is: 1
 
-  Scenario: undefine a relation type throws if instances exist
 
-  Scenario: undefine an attribute type throws if instances exist
+  Scenario: undefining abstract on a type that is already non-abstract does nothing
+    When graql undefine
+      """
+      undefine person abstract;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x type person;
+        not { $x abstract; };
+      get;
+      """
+    When concept identifiers are
+      |     | check | value  |
+      | PER | label | person |
+    Then uniquely identify answer concepts
+      | x   |
+      | PER |
 
-  Scenario: a type can be undefined after deleting all of its instances
 
-  Scenario: undefine 'plays' from type throws if there is an instance playing that role in an existing relation
+  @ignore
+  # TODO fails same as undefine abstract; then require sub-abstract type validation
+  Scenario: undefining a type as abstract errors if has abstract child types (?)
```

Comment:
question mark?

Reply:
and TODOs

Reply:
We decided that this would, in fact, be valid - so I'll implement the scenario, testing that it _doesn't_ error.

Reply:
Done